src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -351,6 +350,7 @@ protected TypeDesc GetDefaultNativeType(NativeType nativeType)
 
                 case MarshallerKind.UnicodeString:
                     return GetDefaultNativeType(NativeType.SysInt); // We pin UniCodeString, so native type will be SysInt
```

Comment:
I tend to think that the GetDefaultNativeType in most of these places just adds unnecessary layer of indirection - makes things harder to understand. E.g. would it be better to just say here:
```
case MarshallerKind.UnicodeString:
      return GetWellKnownType(WellKnownType.Char).MakePointerType();
```

?